C/src/TaskSetGenerator/gui.py

```diff
@@ -0,0 +1,223 @@
+import tkinter as tk
+from tkinter import ttk
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+from scipy.interpolate import interp1d
+import numpy as np
+
+import matplotlib
+matplotlib.use('tkagg')
+
+import os
+import sys
+import subprocess
+import shutil
+from functools import partial
+import random
+
+timeUnits = ['sec', 'msec', 'usec', 'nsec']
+
+def step(num_workers, scheduler_type):
+
+    global char_to_replace, template
+    char_to_replace['SCHEDULER_TYPE'] = scheduler_type
+    char_to_replace['NUM_WORKERS'] = str(num_workers)
+
+    contents = template
+    for key, value in char_to_replace.items():
+        contents = contents.replace(key, value)
+
+    filename = f'practice_{scheduler_type}_{num_workers}'
+    filepath = f'{WORKING_DIR}/.gui/src/{filename}.lf'
+    with open(filepath, "w") as lf_file:
+        lf_file.write(contents)
+        lf_file.close()
+        print(f"File saved: {filepath}")
+
+    out = subprocess.run([f'./gradlew','runLfc', '--args', filepath], capture_output=True)
+
+    print("Build successfully!")
+    lf_out = subprocess.run([f'{WORKING_DIR}/.gui/bin/{filename}'], capture_output=True)
+
+    for line in reversed(lf_out.stdout.decode("utf-8").split('\n')):
+        if line.startswith("---- Elapsed physical"):
+            exe_time = line.split(' ')[-1]
+            break
+
+    exe_time = int(exe_time.replace(',','')) / 1000000000
+    return exe_time
+    
+
+def runLfc():
+
+    global char_to_replace, template
+    workers = [1, 2, 5, 20]
+    schedulers = ["NP", "GEDF_NP", "GEDF_NP_CI"]
```

Comment:
It would be great if the README also included instructions on how and where one could add a new scheduler.

Also, I'm not sure how hard this is but selectable schedulers in the GUI I think would also be useful.